cmd/translate/translate.go

```diff
@@ -0,0 +1,244 @@
+	srcpath := flags.srcpath
+	outpath := flags.outpath
+	name := flags.name
+
+	curate := flags.curate
+	qadisablecli := flags.qadisablecli
+	qaskip := flags.qaskip
+	qaport := flags.qaport
+	qacaches := flags.qacaches
+
+	if planfile, err = filepath.Abs(planfile); err != nil {
+		log.Fatalf("Failed to make the plan file path %q absolute. Error: %q", planfile, err)
+	}
+	if srcpath != "" {
+		if srcpath, err = filepath.Abs(srcpath); err != nil {
+			log.Fatalf("Failed to make the source directory path %q absolute. Error: %q", srcpath, err)
+		}
+	}
+	if outpath, err = filepath.Abs(outpath); err != nil {
+		log.Fatalf("Failed to make the output directory path %q absolute. Error: %q", outpath, err)
+	}
+
+	// Global settings
+	common.IgnoreEnvironment = ignoreEnv
+	qaengine.StartEngine(qaskip, qaport, qadisablecli)
+	cachepaths := []string{}
+	for i := len(qacaches) - 1; i >= 0; i-- {
+		cachepaths = append(cachepaths, qacaches[i])
+	}
+	qaengine.AddCaches(cachepaths)
+
+	// Parameter cleaning and curate plan
+	var p plan.Plan
+	fi, err := os.Stat(planfile)
+	if err == nil && fi.IsDir() {
+		planfile = filepath.Join(planfile, common.DefaultPlanFile)
+		_, err = os.Stat(planfile)
+	}
+	if err != nil {
+		log.Debugf("No plan file found.")
+		if cmd.Flags().Changed(planFlag) || !cmd.Flags().Changed(sourceFlag) {
+			// They specified path to a plan file which doesn't exist.
+			// OR
+			// They didn't specify path to a source directory.
+			log.Fatalf("Error while accessing plan file at path %s Error: %q", planfile, err)
+		}
+		checkSourcePath(srcpath)
+		log.Debugf("Creating a new plan.")
+		p = move2kube.CreatePlan(srcpath, name)
+		outpath = filepath.Join(outpath, p.Name)
+		checkOutputPath(outpath)
+		createOutputDirectoryAndCacheFile(outpath)
+		p = move2kube.CuratePlan(p)
+	} else {
+		log.Infof("Detected a plan file at path %s. Will translate using this plan.", planfile)
+		if p, err = plan.ReadPlan(planfile); err != nil {
+			log.Fatalf("Unable to read the plan at path %s Error: %q", planfile, err)
+		}
+		if len(p.Spec.Inputs.Services) == 0 {
+			log.Fatalf("Failed to find any services. Aborting.")
+		}
+		if cmd.Flags().Changed(nameFlag) {
+			p.Name = name
+		}
+		if cmd.Flags().Changed(sourceFlag) {
+			if err := p.SetRootDir(srcpath); err != nil {
+				log.Fatalf("Failed to set the root directory to %q Error: %q", srcpath, err)
+			}
+		}
+		checkSourcePath(p.Spec.Inputs.RootDir)
+		outpath = filepath.Join(outpath, p.Name)
+		checkOutputPath(outpath)
+		createOutputDirectoryAndCacheFile(outpath)
+		if curate {
+			p = move2kube.CuratePlan(p)
+		}
+	}
+
+	// Translate
+	move2kube.Translate(p, outpath, qadisablecli)
+	log.Infof("Translated target artifacts can be found at [%s].", outpath)
+}
+
+func main() {
+	// Setup
+	must := func(err error) {
+		if err != nil {
+			panic(err)
+		}
+	}
+	viper.AutomaticEnv()
+
+	flags := translateFlags{}
+	translateCmd := &cobra.Command{
+		Use:   "translate",
+		Short: "Translate using move2kube plan",
```

Comment:
We need to change this. When users use plan, they are not using plan directly. Here we are targeting a different usecase of direct translation.